### PR TITLE
Migrate to go mod, one go module per directory

### PR DESCRIPTION
**Pull request by Soulou**

### CODE REVIEW

File: graceful/go.mod [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module github.com/Scalingo/go-utils/graceful
+
+go 1.14
+
+require (
+	github.com/Scalingo/go-utils/logger v0.0.0-00010101000000-000000000000
+	github.com/facebookgo/grace v0.0.0-20180706040059-75cf19382434
+	github.com/stretchr/testify v1.2.2
```
Comment (EtienneM) [inline graceful/go.mod:8]: Update to 1.6.1 as you did on some other packages here?

File: nsqconsumer/go.mod [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	github.com/sirupsen/logrus v1.7.0
+	github.com/stvp/rollbar v0.5.1
+	gopkg.in/errgo.v1 v1.0.1
+)
+
+replace github.com/Scalingo/go-utils/logger => ../logger
+
+replace github.com/Scalingo/go-utils/nsqproducer => ../nsqproducer
+
+replace github.com/Scalingo/go-utils/env => ../env
```
Comment (EtienneM) [inline nsqconsumer/go.mod:18]: It should only be uncommented in dev, shouldn't it?
```suggestion
// replace github.com/Scalingo/go-utils/logger => ../logger
// replace github.com/Scalingo/go-utils/nsqproducer => ../nsqproducer
// replace github.com/Scalingo/go-utils/env => ../env
```

**Review by EtienneM — COMMENTED** [review]
There are a few questions about the dependencies version, otherwise LGTM :)

File: graceful/go.mod [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module github.com/Scalingo/go-utils/graceful
+
+go 1.14
+
+require (
+	github.com/Scalingo/go-utils/logger v0.0.0-00010101000000-000000000000
```
[the remark this answers is not in the document]
Comment (Soulou) [reply graceful/go.mod:6]: Yep I'm doing it :)

File: gomock_generator/go.mod [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module github.com/Scalingo/go-utils/gomock_generator
+
+go 1.14
+
+replace github.com/Scalingo/go-utils/logger => ../logger
+
+require (
+	github.com/Scalingo/go-utils/logger v0.0.0-00010101000000-000000000000
```
[the remark this answers is not in the document]
Comment (Soulou) [reply gomock_generator/go.mod:8]: Yep I'm doing it :)

File: concurrency/go.mod [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module github.com/Scalingo/go-utils/concurrency
+
+go 1.14
+
+require (
+	github.com/Scalingo/go-utils v7.1.0+incompatible
```
[the remark this answers is not in the document]
Comment (Soulou) [reply concurrency/go.mod:6]: Yep I'm doing it :)

File: graceful/go.mod (hunk 1 above)
Comment (Soulou) [reply graceful/go.mod:8]: Yep I'm doing it :)

File: nsqconsumer/go.mod (hunk 2 above)
Comment (Soulou) [reply nsqconsumer/go.mod:18]: Like the others, I forgot some I'll handle it

File: storage/go.mod [hunk 6]
```diff
@@ -0,0 +1,22 @@
+module github.com/Scalingo/go-utils/storage
+
+go 1.14
+
+require (
+	github.com/Scalingo/go-utils/logger v1.0.0 // indirect
+	github.com/aws/aws-sdk-go v1.35.9
+	github.com/aws/aws-sdk-go-v2 v0.27.0
```
Comment (EtienneM) [inline storage/go.mod:8]: I don't understand why we use v1 and v2 of this dependency. Maybe an issue in the code and we should only use v2?

File: storage/go.mod [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+	github.com/aws/aws-sdk-go-v2/credentials v0.1.2
+	github.com/aws/aws-sdk-go-v2/feature/s3/manager v0.1.0
+	github.com/aws/aws-sdk-go-v2/service/internal/accept-encoding v0.2.0 // indirect
+	github.com/aws/aws-sdk-go-v2/service/s3 v0.27.0
+	github.com/awslabs/smithy-go v0.2.0
+	github.com/golang/mock v1.4.4
+	github.com/ncw/swift v1.0.52
+	github.com/pkg/errors v0.9.1
+	github.com/stretchr/testify v1.2.2
```
Comment (EtienneM) [inline storage/go.mod:17]: Update to 1.6.1 as you did on some other packages here?

File: storage/go.mod [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+require (
+	github.com/Scalingo/go-utils/logger v1.0.0 // indirect
+	github.com/aws/aws-sdk-go v1.35.9
+	github.com/aws/aws-sdk-go-v2 v0.27.0
+	github.com/aws/aws-sdk-go-v2/credentials v0.1.2
+	github.com/aws/aws-sdk-go-v2/feature/s3/manager v0.1.0
+	github.com/aws/aws-sdk-go-v2/service/internal/accept-encoding v0.2.0 // indirect
+	github.com/aws/aws-sdk-go-v2/service/s3 v0.27.0
+	github.com/awslabs/smithy-go v0.2.0
```
Comment (EtienneM) [inline storage/go.mod:13]: Update to v0.3.0 which just got released so that we are up-to-date for the v1 release of the storage lib?

**Review by EtienneM — COMMENTED** [review]
I think we should use the latest version of each dependencies when it's possible before release v1 of all these modules. That's why I'm being a bit picky on this review.. :sweat_smile:

File: influx/go.mod [hunk 9]
```diff
@@ -0,0 +1,9 @@
+module github.com/Scalingo/go-utils/influx
+
+go 1.14
+
+require (
+	github.com/influxdata/influxdb v1.8.3
+	github.com/pkg/errors v0.9.1
+	github.com/stretchr/testify v1.4.0
```
[the remark this answers is not in the document]
Comment (Soulou) [reply influx/go.mod]: Done

File: nsqproducer/go.mod [hunk 10]
```diff
@@ -0,0 +1,18 @@
+module github.com/Scalingo/go-utils/nsqproducer
+
+go 1.14
+
+require (
+	github.com/Scalingo/go-utils/env v1.0.1
```
[the remark this answers is not in the document]
Comment (Soulou) [reply nsqproducer/go.mod:6]: Nope I've added a v1.0.1 when I had some issues with the golang module registry

File: storage/go.mod (hunk 7 above)
Comment (Soulou) [reply storage/go.mod]: Done

File: mongo/go.mod [hunk 11]
```diff
@@ -0,0 +1,19 @@
+module github.com/Scalingo/go-utils/mongo
+
+go 1.14
+
+require (
+	github.com/Scalingo/go-utils/errors v1.0.0
+	github.com/Scalingo/go-utils/logger v1.0.0
+	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
+	github.com/sirupsen/logrus v1.7.0
+	github.com/stretchr/testify v1.2.2
```
[the remark this answers is not in the document]
Comment (Soulou) [reply mongo/go.mod]: Done

File: logger/go.mod [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+go 1.14
+
+require (
+	github.com/Scalingo/logrus-rollbar v1.3.2
+	github.com/Soulou/errgo-rollbar v0.1.1 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	github.com/rollbar/rollbar-go v1.2.0
+	github.com/sirupsen/logrus v1.7.0
+	github.com/stretchr/testify v1.2.2
```
[the remark this answers is not in the document]
Comment (Soulou) [reply logger/go.mod]: Done

File: storage/go.mod (hunk 8 above)
Comment (Soulou) [reply storage/go.mod]: Done

**Review by EtienneM — APPROVED** [review]

**Pull request merged**